### PR TITLE
Remove SAS token validation in the SDK

### PR DESCRIPTION
**Pull request by ericwol-msft**
# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/main/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `main` branch. 
  - [ ] I have merged the latest `main` branch prior to submission and re-merged as needed after I took any feedback.
  - [ ] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

**Review by ewertons — APPROVED** [review]

File: iothub_client/src/iothubtransport_amqp_cbs_auth.c [hunk 1]
```diff
@@ -267,31 +267,15 @@ static int create_and_put_SAS_token_to_cbs(AUTHENTICATION_INSTANCE* instance)
...
-                sas_token = NULL;
-                result = MU_FAILURE;
-            }
-            else if (token_status == SAS_TOKEN_STATUS_FAILED)
+            sas_token = IoTHubClient_Auth_Get_SasToken(instance->authorization_module, NULL, 0, NULL);
+            if (sas_token == NULL)
             {
-                LogError("testing Sas Token failed.");
-                sas_token = NULL;
+                LogError("failure getting sas Token.");
```
Comment (danewalton) [inline iothub_client/src/iothubtransport_amqp_cbs_auth.c:273]:
```suggestion
                LogError("Failure getting SAS token.");
```

File: iothub_client/src/iothubtransport_mqtt_common.c [hunk 2]
```diff
@@ -2626,26 +2626,12 @@ static int SendMqttConnectMsg(PMQTTTRANSPORT_HANDLE_DATA transport_data)
...
-        {
-            transport_data->transport_callbacks.connection_status_cb(IOTHUB_CLIENT_CONNECTION_UNAUTHENTICATED, IOTHUB_CLIENT_CONNECTION_EXPIRED_SAS_TOKEN, transport_data->transport_ctx);
-            result = MU_FAILURE;
-        }
-        else if (token_status == SAS_TOKEN_STATUS_FAILED)
+        sasToken = IoTHubClient_Auth_Get_SasToken(transport_data->authorization_module, NULL, 0, NULL);
+        if (sasToken == NULL)
         {
-            transport_data->transport_callbacks.connection_status_cb(IOTHUB_CLIENT_CONNECTION_UNAUTHENTICATED, IOTHUB_CLIENT_CONNECTION_BAD_CREDENTIAL, transport_data->transport_ctx);
+            LogError("failure getting sas Token.");
```
Comment (danewalton) [inline iothub_client/src/iothubtransport_mqtt_common.c:2632]: Same here

**Review by danewalton — APPROVED** [review]

File: iothub_client/src/iothubtransport_mqtt_common.c (hunk 2 above)
Comment (ericwol-msft) [reply iothub_client/src/iothubtransport_mqtt_common.c:2632]: delete line 2637???

Comment (danewalton) [reply iothub_client/src/iothubtransport_mqtt_common.c:2632]: Sorry no same as the previous change for the log error. Capitalization

Comment (CIPop) [reply iothub_client/src/iothubtransport_mqtt_common.c:2632]: Nit: changing log may have adverse effects on automatic log scanners.
I don't think we've ever guaranteed log backward compatibility, just something to consider, especially given our design that won't report different error codes (i.e. `MU_FAILURE` is set to indicate a line-number and will change in value with this modification).

**Review by CIPop — APPROVED** [review]
Thanks @ericwol-msft !

**Pull request merged**